addons/website_sale_slides/models/slide_channel.py

```diff
@@ -52,10 +52,19 @@ def write(self, vals):
         return res
 
     def _synchronize_product_publish(self):
+        """
+        Ensure that when publishing a course that its linked product is also published
+        If all courses linked to a product are unpublished, we also unpublished the product
+        """
         if not self:
             return
         self.filtered(lambda channel: channel.is_published and not channel.product_id.is_published).sudo().product_id.write({'is_published': True})
-        self.filtered(lambda channel: not channel.is_published and channel.product_id.is_published).sudo().product_id.write({'is_published': False})
+
+        unpublished_channel_products = self.filtered(lambda channel: not channel.is_published).product_id
+        for product in unpublished_channel_products:
+            linked_courses = self.search([('product_id', '=', product.id)])
+            if all(not course.is_published for course in linked_courses):
```

Comment:
This creates a search in a loop, not great.

We could do a read_group grouped on product_id, and only search for "is_publish = True" in the domain.
Then if you find a count of 0, it means you can unpublish the product.
Do you see what I mean?

(This may require to flush the "is_published" field, not sure, test it without flushing first).

Side-note: same idea about the write, should not be called in a loop